compiler/env/CPU.hpp

```diff
@@ -26,11 +26,20 @@
 
 namespace TR
 {
-class CPU : public OMR::CPUConnector
+class OMR_EXTENSIBLE CPU : public OMR::CPUConnector
    {
    public:
 
+   /** 
+    * @brief Default constructor that defaults down to OMR minimum supported CPU and features
+    */
    CPU() : OMR::CPUConnector() {}
+   
+   /** 
+    * @brief Constructor that initializes the cpu from processor description provided by user
+    * @param[in] OMRProcessorDesc : the input processor description
+    */
+   CPU(const OMRProcessorDesc& processorDescription) : OMR::CPUConnector(processorDescription) {}
```

Comment:
The documentation here should likely go into OMRCPU.hpp as that is the base extensible class.